src/database/database.cc

```diff
@@ -44,20 +44,18 @@ Database::Database(std::shared_ptr<Config> config)
 std::shared_ptr<Database> Database::createInstance(const std::shared_ptr<Config>& config, const std::shared_ptr<Timer>& timer)
 {
     std::string type = config->getOption(CFG_SERVER_STORAGE_DRIVER);
-    std::shared_ptr<Database> database;
-
-    if (type == "sqlite3") {
-        database = std::static_pointer_cast<Database>(std::make_shared<Sqlite3Database>(config, timer));
-    }
+    auto database = [&] {
```

Comment:
Personally I think the old code is clearer

Reply:
The goal here is to avoid database ever being NULL as that results in warnings.

Reply:
I redid this slightly.